docs/src/ring_interface.md

```diff
@@ -846,7 +846,7 @@ parent_type(::Type{ConstPoly{T}}) where T <: RingElement = ConstPolyRing{T}
 
 elem_type(::Type{ConstPolyRing{T}}) where T <: RingElement = ConstPoly{T}
 
-base_ring_type(::Type{ConstPoly{T}}) where T <: RingElement = parent_type(T)
+base_ring_type(::Type{ConstPolyRing{T}}) where T <: RingElement = parent_type(T)
 
 base_ring(R::ConstPolyRing) = R.base_ring::base_ring_type(R)
```

Comment:
To me it seems like we rather need `base_ring_type(::ConstPolyRing{T})` here.

Reply:
The line https://github.com/Nemocas/AbstractAlgebra.jl/blob/2235d6a60348cbb60f4af97332a0dbe49ae24a51/src/fundamental_interface.jl#L127 redirects to the call with the type. So implementing `base_ring_type(::Type{Parenting})` is indeed enough, as the rest is handled by generic code

Reply:
Hm, weird... I can only get `base_ring` for my custom ring to work when implementing `base_ring_type` directly. Here is my (work in progress) implementation: https://github.com/oscar-system/GenericCharacterTables.jl/pull/82

Reply:
As @fingolfin already pointed out over there, this is due to a missing import. (x-ref https://github.com/oscar-system/GenericCharacterTables.jl/pull/82/files#r1689365997)